### DISCUSSION TITLE
Using DP to fit charge density distribution, a problem related to accuracy

### DISCUSSION THREAD

**Discussion by theAfish** [deepmodeling/deepmd-kit#2693] [Q&A]
Hi all,

This is actually related to my Hackathon 2023 project. I am recently trying to fit charge density distribution with deepmd-kit. I have written the preprocessing tools to transform CHGCARs into datasets for atomic energy fitting in deepmd-kit. The fitting for total charge density of single crystal Si and bcc Fe are good, but the results for amorphous Si (understandable) and spin-polarized charge density of Fe are bad, as shown in the figure below, all the tests are based on validation sets:

<img src="https://github.com/deepmodeling/deepmd-kit/assets/85424334/2ef4ea48-90e2-42b0-8466-e7977cf5255e" width=50% height=50%>

and here's the charge density prediction of DP and DFT for amorphous Si:
![image](https://github.com/deepmodeling/deepmd-kit/assets/85424334/a5ebf91d-28bb-4461-b39c-8eaaca12b121)

The input.json is attached here: [input.txt](https://github.com/deepmodeling/deepmd-kit/files/12143403/input.txt) (Fe and Si are similar)

I have tried expand the network complexity, activation function, training steps, etc. The results changed little, as shown below for Si:

![image](https://github.com/deepmodeling/deepmd-kit/assets/85424334/0a9fbcfc-c4e3-4767-96eb-820a42e90020)
![image](https://github.com/deepmodeling/deepmd-kit/assets/85424334/7995d1bf-7f50-404e-9287-b596de21c109)
<img src="https://github.com/deepmodeling/deepmd-kit/assets/85424334/7ba9d7c3-6ed4-40a1-9a18-02b687dd15ac" width=50% height=50%>

So here are my questions:
1. The accuracy decrease for Si is obviously caused by structure complexity. But I've read the deepmd-kit's paper, and I believe that the embedding matrix could grab all the environment information preserving symmetry. If so, why the changing of network structure have little impact on the results?
2.  The accuracy for spin charge density (up-down) for Fe is worse than the total charge density, and I am also trying to find a way to increase the accuracy. Here are the [INCAR](https://github.com/deepmodeling/deepmd-kit/files/12143562/INCAR.txt) and [KPOINTS](https://github.com/deepmodeling/deepmd-kit/files/12143563/KPOINTS.txt) used for Fe.

I was wondering whether anyone can give any techical point of view of these problems. And I would like to provide more details.

Thank you very much!

**Comment by AnguseZhang** [comment]
1. For amorphous Si, it's possible that the fitting accuracy is poorer compared to crystal Si since in amorphous environment the magnitude of forces is larger as well. Whether the current fitting is enough? It depends on the final purpose of your project. I also suggest that you try to provision and figure out how to utilize the trained model (just like training a deep potential and then pack it with LAMMPS to perform molecular dynamics and caculate a series of physical properties).
2. For spin-polarized charge density of Fe, have you even learned about DeepSPIN, a solution that can handle with spin-polarized systems?

**Reply by theAfish** [reply]
Thank you for your reply!

1. The lcurves tend to be horizontal at the end of each training.  There is a slight downward trend for more complex networks, so I'm currently testing longer training.
2. The purpose of this model is also related to its accuracy. If the accuracy is relatively low, it could provide coarse-grained charge distributions or charge states transitions in interesting regions of MD simulations. If it has higher accuracy, other properties such as exciton, polaron, etc. can be obtained. I was also wondering about the possibility or difficulty of providing the charge information using the particle-mesh method for MD simulations.
3. Yes, I know the DeepSPIN model, but I haven't used it. Here the reason I used spin-charge density is to test whether the model can handle the spin-influenced charge density/charge difference since they can be derived directly and are more complex than the total charge density. Other scalar fields such as electron localization function(ELF) could also be used.
